ui/appui-react/src/appui-react/UiFramework.ts

```diff
@@ -854,4 +906,354 @@ export class UiFramework {
+      el,
+      location,
+      offset,
+      onItemExecuted,
+      onCancel,
+      relativePosition
+    );
+  }
+
+  /**
+   * Hides a Card displayed with {@link UiFramework.showCard}
+   */
+  public static hideCard() {
+    PopupManager.hideCard();
+  }
+
+  /** Opens a Tool Settings Ui popup at a particular location.
+   * @param dataProvider The DialogLayoutDataProvider for the tool settings popup dialog.
+   * @param location Location of the tool settings, relative to the origin of anchorElement or the window
+   * @param offset Offset of the tool settings from the location
+   * @param onCancel Function invoked when the Escape key is pressed or a click occurs outside the tool settings
+   * @param relativePosition Position relative to the given location. Defaults to TopRight.
+   * @param anchorElement The HTMLElement that anchors the tool settings. If undefined, the location is relative to the overall window.
+   * @return true if the tool settings were displayed, false if the tool settings could not be displayed.
+   */
+  public static openToolSettingsPopup(
+    dataProvider: DialogLayoutDataProvider,
+    location: XAndY,
+    offset: XAndY,
+    onCancel: OnCancelFunc,
+    relativePosition?: RelativePosition,
+    anchorElement?: HTMLElement
+  ): boolean {
+    const el = this.resolveHtmlElement(anchorElement);
+
+    if (relativePosition === undefined)
+      relativePosition = RelativePosition.TopRight;
+
+    return PopupManager.openToolSettings(
+      dataProvider,
+      el,
+      location,
+      offset,
+      onCancel,
+      relativePosition
+    );
+  }
+
+  /**
+   * Closes the Tool Settings Ui popup.
+   */
+  public static closeToolSettingsPopup() {
+    PopupManager.closeToolSettings();
+  }
+
+  /** Show a Toolbar at a particular location.
+   * @param toolbarProps Properties of the Toolbar to display.
+   * @param location Location of the Toolbar, relative to the origin of anchorElement or the overall window.
+   * @param offset Offset of the Toolbar from the location.
+   * @param onItemExecuted Function invoked after a Toolbar item is executed
+   * @param onCancel Function invoked when the Escape key is pressed or a click occurs outside the Toolbar
+   * @param relativePosition Position relative to the given location. Defaults to TopRight.
+   * @param anchorElement The HTMLElement that anchors the Toolbar. If undefined, the location is relative to the overall window.
+   * @return true if the Toolbar was displayed, false if the Toolbar could not be displayed.
+   */
+  public static showToolbar(
+    toolbarProps: AbstractToolbarProps,
+    location: XAndY,
+    offset: XAndY,
+    onItemExecuted: OnItemExecutedFunc,
+    onCancel: OnCancelFunc,
+    relativePosition?: RelativePosition,
+    anchorElement?: HTMLElement
+  ): boolean {
+    if (relativePosition === undefined)
+      relativePosition = RelativePosition.TopRight;
+
+    const el = UiFramework.resolveHtmlElement(anchorElement)
```

Comment:
```suggestion
    relativePosition: RelativePosition = RelativePosition.TopRight,
    anchorElement?: HTMLElement
  ): boolean {
    const el = UiFramework.resolveHtmlElement(anchorElement)
```
Can be replaced with an initializer.